File: Backend/Models/Pedido.cs
using Backend.Models.Enums;

namespace Backend.Models;

public class Pedido
{
    public int Id { get; set; }
    public int ClienteId { get; set; }
    public DateTime DataCriacao { get; set; } = DateTime.UtcNow;
    public DateTime? DataEntrega { get; set; }
    public decimal ValorTotal { get; set; }
    public StatusPedido Status { get; set; } = StatusPedido.Pendente;
    public string? Observacoes { get; set; }

    // Navigation
    public Cliente Cliente { get; set; } = null!;
    public ICollection<ItemPedido> Itens { get; set; } = new List<ItemPedido>();
}

File: backend/src/models/Pedido.ts
import { StatusPedido } from './enums';

/**
 * Interface que representa uma linha da tabela `pedidos`.
 */
export interface Pedido {
  id: number;
  cliente_id: number;
  data_criacao: string;
  data_entrega: string | null;
  valor_total: number;
  status: StatusPedido;
  observacoes: string | null;
}
